src/blog/2024/04/using-webhook-with-node-red.md

```diff
@@ -0,0 +1,188 @@
+
+It's common and understandable to get confused between APIs and webhooks, especially when you are learning about webhooks for the first time. However, comparing the two can help dispel these confusions.
+
+## API:
+
+- **Direction:**
+  - Typically involves client-to-server communication.
+- **Initiation:**
+  - Requests are initiated by the client.
+- **Request Method:**
+  - Usually employs HTTP methods like GET, POST, etc.
+- **Response:**
+  - Provides an immediate response upon request.
+- **Data Transfer:**
+  - Utilizes a pull model where the client fetches data.
+- **Polling:**
+  - Requires periodic polling for updates.
+- **Payload:**
+  - The client specifies the payload in the request.
+- **Error Handling:**
+  - Typically includes error codes and messages.
+
+## Webhook:
+
+- **Direction:**
+  - Typically involves server-to-server communication.
+- **Initiation:**
+  - Requests are initiated by the server.
+- **Request Method:**
+  - Typically uses the HTTP POST method.
+- **Response:**
+  - Does not provide an immediate response; asynchronous.
+- **Data Transfer:**
+  - Operates on a push model where the server pushes data to the client.
+- **Polling:**
+  - No need for polling; receives updates directly.
+- **Payload:**
+  - The server defines the payload in the outgoing request.
+- **Error Handling:**
+  - Errors are handled by retry mechanisms or manual intervention.
+
+### Example Scenario:
+
+For example, Consider a manufacturing facility that utilizes temperature sensors to monitor temperature levels in critical areas. When the temperature surpasses predefined thresholds, it triggers a series of actions for maintenance and monitoring.
+
+!["Diagram explaining how component works in"](./images/using-webhook-with-node-red-diagram.png "Role Based Access Control For Node-RED with FlowFuse"){data-zoomable}
+
+- Temperature Sensor (Server 1): These physical sensors are installed in the manufacturing plant to monitor temperature. They trigger webhook requests to Server 2 whenever abnormal temperature patterns are detected.
+
+- Server 2 (Webhook Server): This server creates and hosts the webhook endpoint. It receives HTTP requests from the temperature sensors (Server 1) when abnormal temperatures are detected. The request contains temperature data. Server 2 then processes this data and sends a POST request with relevant information to Server 3.
+
+- Maintenance System (Server 3): This system receives POST requests from Server 2 containing event-related data on a specific endpoint provided to Server 2. It then automatically schedules maintenance tasks based on the received information.
+
+## Practical implementation
+
+In this section, we will construct the practical implementation of the scenario described above. However, all three components or servers will be hosted on the same Node-RED instance in our example. Furthermore, we will simulate data rather than using real sensors by utilizing random number expressions.
```

Comment:
I think this is cheating. I think you have a RPI, can you install the device agent on it and add a temperature sensor to it? You can expense both if you don't have it yet.

Than setup the device agent of Server 1, connect the sensor. Don't use Random

Reply:
Currently, I don't have an RPI or related equipment, but I'll purchase them as soon as possible. I'll make sure to update this blog and incorporate it into our next post.

Reply:
Now you've got your RPI, please let me know when this post is ready :)